code/modules/recycling/disposal/eject.dm

```diff
@@ -1,23 +1,13 @@
-// called when movable is expelled from a disposal pipe or outlet
-// by default does nothing, override for special behaviour
-
-/atom/movable/proc/pipe_eject(direction)
-	return
-
-/obj/effect/decal/cleanable/blood/gibs/pipe_eject(direction)
-	var/list/dirs
-	if(direction)
-		dirs = list(direction, turn(direction, -45), turn(direction, 45))
-	else
-		dirs = GLOB.alldirs.Copy()
-
-	streak(dirs)
-
-/obj/effect/decal/cleanable/robot_debris/gib/pipe_eject(direction)
-	var/list/dirs
-	if(direction)
-		dirs = list(direction, turn(direction, -45), turn(direction, 45))
-	else
-		dirs = GLOB.alldirs.Copy()
-
-	streak(dirs)
+/**
+  * General proc used to expel an holder's contents through src (for bins, holder is also the src).
+  * Because having more than one source of a determinated signal is usually bad practice.
```

Comment:
```suggestion
```
This comment won't really age well, it's only contextual to this PR but won't be informative for future coders.
Yes, we want abstract events reported and handled generically, instead of repeated code.